code/modules/surgery/organs/organ_internal.dm

```diff
@@ -60,6 +60,13 @@
 
 //Special is for instant replacement like autosurgeons
 /obj/item/organ/proc/Remove(mob/living/carbon/M, special = FALSE)
+	//Stop any reagent metabolizing on organ destruction
+	if(reagents && iscarbon(owner))
+		var/mob/living/carbon/body = owner
+		for(var/bit in reagents.reagent_list)
+			var/datum/reagent/_reagent = bit
+			_reagent.on_mob_end_metabolize(body)
```

Comment:
and what if that reagent already wasn't being metabolized, due to the absence of a liver or the like? can I use this to repeatedly gain on_mob_end_metabolize() effects?

Reply:
yep you are not wrong, will get this fixed up.